### PR TITLE
[WFLY-17868] Replace uses of ServiceName SocketBinding.JBOSS_BINDING_…

### PR DESCRIPTION
**Pull request by soul2zimate**
…NAME with use of the capability in RTS subsystem

Issue: https://issues.redhat.com/browse/WFLY-17868

### CODE REVIEW

**Review by tadamski — APPROVED** [review]

File: rts/src/main/java/org/wildfly/extension/rts/RTSSubsystemAdd.java [hunk 1]
```diff
@@ -97,16 +105,13 @@ private void registerCoordinatorService(final OperationContext context, final Mo
...
-                .addDependency(SocketBinding.JBOSS_BINDING_NAME.append(socketBindingName), SocketBinding.class,
-                        coordinatorService.getInjectedSocketBinding())
-                .addDependency(UndertowService.virtualHostName(serverName, hostName), Host.class,
-                        coordinatorService.getInjectedHost());
-
-        coordinatorServiceBuilder
+        final CapabilityServiceBuilder<?> builder = context.getCapabilityServiceTarget().addCapability(RTS_COORDINATOR_CAPABILITY);
+        Supplier<Host> hostSupplier = builder.requires(UndertowService.virtualHostName(serverName, hostName));
+        ServiceName serviceName = context.getCapabilityServiceName(SOCKET_BINDING_CAPABILITY_NAME, socketBindingName, SocketBinding.class);
+        Supplier<SocketBinding> socketBindingSupplier = builder.requires(serviceName);
```
Comment (pferraro) [inline rts/src/main/java/org/wildfly/extension/rts/RTSSubsystemAdd.java:111]:
```java
Supplier<SocketBinding> socketBindingSupplier = builder.requiresCapability(SOCKET_BINDING_CAPABILITY_NAME, SocketBinding.class, socketBindingName);

File: rts/src/main/java/org/wildfly/extension/rts/RTSSubsystemAdd.java [hunk 2]
```diff
@@ -116,37 +121,31 @@ private void registerParticipantService(final OperationContext context, final Mo
...
-                .addDependency(SocketBinding.JBOSS_BINDING_NAME.append(socketBindingName), SocketBinding.class,
-                        participantService.getInjectedSocketBinding())
-                .addDependency(UndertowService.virtualHostName(serverName, hostName), Host.class,
-                        participantService.getInjectedHost());
 
-        participantServiceBuilder
+        final CapabilityServiceBuilder<?> builder = context.getCapabilityServiceTarget().addCapability(RTS_PARTICIPANT_CAPABILITY);
+        Supplier<Host> hostSupplier = builder.requires(UndertowService.virtualHostName(serverName, hostName));
+        ServiceName serviceName = context.getCapabilityServiceName(SOCKET_BINDING_CAPABILITY_NAME, socketBindingName, SocketBinding.class);
+        Supplier<SocketBinding> socketBindingSupplier = builder.requires(serviceName);
```
Comment (pferraro) [inline rts/src/main/java/org/wildfly/extension/rts/RTSSubsystemAdd.java:128]:
```java
Supplier<SocketBinding> socketBindingSupplier = builder.requiresCapability(SOCKET_BINDING_CAPABILITY_NAME, SocketBinding.class, socketBindingName);

File: rts/src/main/java/org/wildfly/extension/rts/RTSSubsystemAdd.java [hunk 3]
```diff
@@ -116,37 +121,31 @@ private void registerParticipantService(final OperationContext context, final Mo
...
-                .addDependency(SocketBinding.JBOSS_BINDING_NAME.append(socketBindingName), SocketBinding.class,
-                        volatileParticipantService.getInjectedSocketBinding())
-                .addDependency(UndertowService.virtualHostName(serverName, hostName), Host.class,
-                        volatileParticipantService.getInjectedHost());
 
-        volatileParticipantServiceBuilder
+        final CapabilityServiceBuilder<?> builder = context.getCapabilityServiceTarget().addCapability(RTS_VOLATILE_PARTICIPANT_CAPABILITY);
+        Supplier<Host> hostSupplier = builder.requires(UndertowService.virtualHostName(serverName, hostName));
+        ServiceName serviceName = context.getCapabilityServiceName(SOCKET_BINDING_CAPABILITY_NAME, socketBindingName, SocketBinding.class);
+        Supplier<SocketBinding> socketBindingSupplier = builder.requires(serviceName);
```
Comment (pferraro) [inline rts/src/main/java/org/wildfly/extension/rts/RTSSubsystemAdd.java:145]:
```java
Supplier<SocketBinding> socketBindingSupplier = builder.requiresCapability(SOCKET_BINDING_CAPABILITY_NAME, SocketBinding.class, socketBindingName);

**Review by pferraro — CHANGES_REQUESTED** [review]

**Comment by soul2zimate** [comment]
Updated, thanks.

File: rts/src/main/java/org/wildfly/extension/rts/RTSSubsystemAdd.java [hunk 4]
```diff
@@ -97,16 +104,12 @@ private void registerCoordinatorService(final OperationContext context, final Mo
...
-                .getServiceTarget()
-                .addService(RTSSubsystemExtension.COORDINATOR, coordinatorService)
-                .addDependency(SocketBinding.JBOSS_BINDING_NAME.append(socketBindingName), SocketBinding.class,
-                        coordinatorService.getInjectedSocketBinding())
-                .addDependency(UndertowService.virtualHostName(serverName, hostName), Host.class,
-                        coordinatorService.getInjectedHost());
-
-        coordinatorServiceBuilder
+        final CapabilityServiceBuilder<?> builder = context.getCapabilityServiceTarget().addCapability(RTS_COORDINATOR_CAPABILITY);
+        Supplier<Host> hostSupplier = builder.requires(UndertowService.virtualHostName(serverName, hostName));
```
Comment (pferraro) [inline rts/src/main/java/org/wildfly/extension/rts/RTSSubsystemAdd.java:108]: This should also use a capability reference, but we can address separately.  Can you file a jira?

File: rts/src/main/java/org/wildfly/extension/rts/RTSSubsystemAdd.java [hunk 5]
```diff
@@ -116,37 +119,29 @@ private void registerParticipantService(final OperationContext context, final Mo
...
-                .getServiceTarget()
-                .addService(RTSSubsystemExtension.PARTICIPANT, participantService)
-                .addDependency(SocketBinding.JBOSS_BINDING_NAME.append(socketBindingName), SocketBinding.class,
-                        participantService.getInjectedSocketBinding())
-                .addDependency(UndertowService.virtualHostName(serverName, hostName), Host.class,
-                        participantService.getInjectedHost());
 
-        participantServiceBuilder
+        final CapabilityServiceBuilder<?> builder = context.getCapabilityServiceTarget().addCapability(RTS_PARTICIPANT_CAPABILITY);
+        Supplier<Host> hostSupplier = builder.requires(UndertowService.virtualHostName(serverName, hostName));
```
Comment (pferraro) [inline rts/src/main/java/org/wildfly/extension/rts/RTSSubsystemAdd.java:124]: Same as above.

File: rts/src/main/java/org/wildfly/extension/rts/RTSSubsystemAdd.java [hunk 6]
```diff
@@ -116,37 +119,29 @@ private void registerParticipantService(final OperationContext context, final Mo
...
-                .getServiceTarget()
-                .addService(RTSSubsystemExtension.VOLATILE_PARTICIPANT, volatileParticipantService)
-                .addDependency(SocketBinding.JBOSS_BINDING_NAME.append(socketBindingName), SocketBinding.class,
-                        volatileParticipantService.getInjectedSocketBinding())
-                .addDependency(UndertowService.virtualHostName(serverName, hostName), Host.class,
-                        volatileParticipantService.getInjectedHost());
 
-        volatileParticipantServiceBuilder
+        final CapabilityServiceBuilder<?> builder = context.getCapabilityServiceTarget().addCapability(RTS_VOLATILE_PARTICIPANT_CAPABILITY);
+        Supplier<Host> hostSupplier = builder.requires(UndertowService.virtualHostName(serverName, hostName));
```
Comment (pferraro) [inline rts/src/main/java/org/wildfly/extension/rts/RTSSubsystemAdd.java:140]: Same as above.

**Review by pferraro — APPROVED** [review]

File: rts/src/main/java/org/wildfly/extension/rts/RTSSubsystemAdd.java (hunk 4 above)
Comment (soul2zimate) [reply rts/src/main/java/org/wildfly/extension/rts/RTSSubsystemAdd.java:108]: Created https://issues.redhat.com/browse/WFLY-17882

**Review by pferraro — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks @soul2zimate, @pferraro and @tadamski

**Pull request merged**